### PR TITLE
Field mapping implementation

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

File: src/app/core/models/enum/enum.model.ts [hunk 1]
```diff
@@ -99,4 +99,10 @@ export enum QBDOnboardingState {
   export enum QBDEntity {
     EMPLOYEE = 'EMPLOYEE',
     VENDOR = 'VENDOR'
+  }
+
+  export enum QBDRepresentation {
+    PROJECT = 'PROJECT',
+    COST_CENTER = 'COST_CENTER',
+    CUSTOME_FIELD = 'CUSTOME_FIELD'
```
Comment (ashwin1111) [inline src/app/core/models/enum/enum.model.ts:107]: CUSTOMER

Comment (ashwin1111) [reply src/app/core/models/enum/enum.model.ts:107]: this looks like fyle fileds, let's name it `FyleField`

Comment (ashwin1111) [reply src/app/core/models/enum/enum.model.ts:107]: remove customer, looks like a custom field

File: src/app/core/models/qbd/qbd-configuration/field-mapping.model.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+export class FieldMappingModel {
+    static constructPayload(fieldMappingForm: FormGroup): QBDFieldMappingPost {
+      const fieldMappingPayload: QBDFieldMappingPost = {
+        class_type: fieldMappingForm.get('classType')?.value ? fieldMappingForm.get('classType')?.value : null,
+        project_type: fieldMappingForm.get('customerType')?.value ? fieldMappingForm.get('customerType')?.value : null
+
+      };
+      return fieldMappingPayload;
+    }
+    }
```
Comment (ashwin1111) [inline src/app/core/models/qbd/qbd-configuration/field-mapping.model.ts:26]: indent is off

File: src/app/shared/components/qbd/configuration/field-mapping/field-mapping.component.ts [hunk 3]
```diff
@@ -1,15 +1,100 @@
...
+      label: 'Select type custom field',
+      value: QBDRepresentation.CUSTOME_FIELD
+    }
+  ];
 
   ngOnInit(): void {
+    this.getSettingsAndSetupForm();
+  }
+
+  constructPayloadAndSave(): void {
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/configuration/field-mapping/field-mapping.component.ts:57]: all functions should be above ngOnInit()

File: src/app/shared/components/qbd/configuration/field-mapping/field-mapping.component.ts [hunk 4]
```diff
@@ -1,15 +1,100 @@
...
+
+  save(): void {
+    if (this.fieldMappingForm.valid) {
+      this.constructPayloadAndSave();
+    }
   }
 
+  private getSettingsAndSetupForm(): void {
+    this.isOnboarding = this.router.url.includes('onboarding');
+    this.fieldMappingService.getQbdFieldMapping().subscribe((fielsMappingResponse : QBDFieldMappingGet) => {
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/configuration/field-mapping/field-mapping.component.ts:82]: `fieldMappingResponse`

File: src/app/shared/components/qbd/configuration/field-mapping/field-mapping.component.ts [hunk 5]
```diff
@@ -1,15 +1,100 @@
...
+      this.fieldMapping = fielsMappingResponse;
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/configuration/field-mapping/field-mapping.component.ts:83]: where is this used?

File: src/app/shared/components/qbd/configuration/field-mapping/field-mapping.component.ts [hunk 6]
```diff
@@ -1,15 +1,100 @@
...
+      this.fieldMappingForm = this.formBuilder.group({
+        classType: [this.fieldMapping?.class_type ? this.fieldMapping?.class_type : null],
+        customerType: [this.fieldMapping?.project_type ? this.fieldMapping?.project_type : null]
+      });
+      this.isLoading = false;
+    }, () => {
+        this.fieldMappingForm = this.formBuilder.group({
+          classType: [null, Validators.required],
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/configuration/field-mapping/field-mapping.component.ts:91]: not a mandatory field, confirmed on slack

File: src/app/shared/components/qbd/configuration/field-mapping/field-mapping.component.ts [hunk 7]
```diff
@@ -1,15 +1,100 @@
...
+    {
+      label: 'Project',
+      value: QBDRepresentation.PROJECT
+    },
+    {
+      label: 'Cost Center',
+      value: QBDRepresentation.COST_CENTER
+    },
+    {
+      label: 'Select type custom field',
```
Comment (ashwin1111) [inline src/app/shared/components/qbd/configuration/field-mapping/field-mapping.component.ts:48]: remove this option, API support is not live

File: src/app/shared/components/qbd/configuration/field-mapping/field-mapping.component.ts (hunk 5 above)
Comment (DhaaraniCIT) [reply src/app/shared/components/qbd/configuration/field-mapping/field-mapping.component.ts:83]: for constructing the form

Comment (ashwin1111) [reply src/app/shared/components/qbd/configuration/field-mapping/field-mapping.component.ts:83]: can you assign it to const? global level is not needed

**Review by ashwin1111 — APPROVED** [review]

File: src/app/core/models/enum/enum.model.ts [hunk 8]
```diff
@@ -99,4 +99,9 @@ export enum QBDOnboardingState {
   export enum QBDEntity {
     EMPLOYEE = 'EMPLOYEE',
     VENDOR = 'VENDOR'
+  }
+
+  export enum QBDRepresentation {
```
Comment (ashwin1111) [inline src/app/core/models/enum/enum.model.ts:104]: name it `FyleField`

**Pull request merged**